lib/rules/no-test-module-for.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const types = require('../utils/types');
+const emberUtils = require('../utils/ember');
+
+const ERROR_MESSAGE = 'moduleFor* apis are are not allowed.';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Disallow use of moduleFor, moduleForComponent, etc',
+      category: 'Testing',
+      recommended: false,
+    },
+    fixable: null,
+  },
+
+  ERROR_MESSAGE,
+
+  create(context) {
+    const filename = context.getFilename();
+    const isTestFile =
+      emberUtils.isTestFile(filename) ||
+      (filename.includes('tests') && filename.includes('helpers'));
+
+    if (!isTestFile) {
+      return {};
+    }
+
+    return {
+      FunctionDeclaration(node) {
+        const id = node.id;
+        if (types.isIdentifier(id) && id.name.startsWith('moduleFor')) {
```

Comment:
As a final condition, can we also use the `getSourceModuleNameForIdentifier` util to ensure this function comes from `qunit`?

Reply:
moduleForAcceptance can be named anything since it's a default export. Like, `moduleFor`, is there a good way to handle that?
looks like I can only handle unit/model and rendering tests with that technique, per: https://github.com/emberjs/ember-qunit/blob/master/docs/legacy.md

Reply:
This is what I was suggesting, would this work?

```
getSourceModuleNameForIdentifier(context, id, 'ember-qunit')
```

Reply:
so, something like this:
```
let isModuleFor = (
  getSourceModuleNameForIdentifier(context, id, 'ember-qunit').includes('moduleFor')
  || types.isIdentifier(id) && id.name.includes('moduleFor')
);
```
?

Reply:
Sorry I made a typo. I meant:

```
if (
  types.isIdentifier(id) &&
  id.name.startsWith('moduleFor') &&
  getSourceModuleNameForIdentifier(context, node.superClass) === 'ember-qunit'
) {
  // report violation
}
```

All `getSourceModuleNameForIdentifier` does is check that the function comes from the right import location.

Reply:
how do we handle moduleForAcceptance?

Reply:
Oh, so if `moduleForAcceptance` can be imported from anywhere, how about skipping the import check for that one?

```
if (
  types.isIdentifier(id) &&
  id.name === 'moduleForAcceptance' ||
  (
    id.name.startsWith('moduleFor') &&
    getSourceModuleNameForIdentifier(context, id) === 'ember-qunit'
  )
) {
  // report violation
}
```
Would this set of checks be accurate?

Reply:
it'd cover the majority of cases. Though, if someone misspelled moduleForAcceptance, it wouldn't include that. I think the default location for that is `app-name/tests/helpers/start-app.js`? maybe? idk. Not sure if that's a standard or not.

Reply:
the issue with moduleForAcceptance, is that it's user/blueprint defined per-app.

Reply:
I see. So are you planning to try adding the logic I suggested? Based on what you said, it sounds like my suggested logic would work. If there's still a problem, let me know.